airframe-sql/src/main/scala/wvlet/airframe/sql/model/LogicalPlan.scala

```diff
@@ -774,7 +775,29 @@ object LogicalPlan {
     override def outputAttributes: Seq[Attribute] = mergeOutputAttributes
     protected def mergeOutputAttributes: Seq[Attribute] = {
       // Collect all input attributes
-      val outputAttributes: Seq[Seq[Attribute]] = children.map(_.outputAttributes.flatMap(_.inputColumns))
+      def collectInputAttributes(rels: Seq[Relation], result: Seq[Seq[Attribute]] = Nil): Seq[Seq[Attribute]] = {
+        rels.flatMap {
+          case s: SetOperation => collectInputAttributes(s.children, result)
```

Comment:
btw, which part is setting the result other than Nil? It looks like this argument is unnecessary.

Reply:
Oh, that's right.